### DISCUSSION TITLE
php:7.4-apache

### DISCUSSION THREAD

**Discussion by adil-azmy** [mlocati/docker-php-extension-installer#587] [Q&A] [answered]
How to make this work with php:7.4-apache?

Seems that dockerfile fail installing extensions without any error.

thank you

**Comment by mlocati** [comment]
Are you sure it fails? Maybe you only have to restart apache?

**Reply by adil-azmy** [reply]
I use this to buil a new image in a dockerfile the container is still not starting apache service at that point no?
```
FROM php:7.4-apache

ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/

RUN \
    # Set -ex and chain commands using ';' instead of '&&'.
    set -ex; \
    \
    apt-get update; \
    \
    chmod +x /usr/local/bin/install-php-extensions; \
    \
    install-php-extensions mysqli 
```

1. even so, i get into docker container and run php -m after starting... no module from the list
2. restarted apach service... still nothing
3. inside the container, tried to manually fire `install-php-extensions mysqli `on an extension, restarted apache, and there it works and installes mysqli.

**Comment by mlocati** [comment, marked as answer]
1. I created a Dockerfile like the one you provided:
   ```Dockerfile
   FROM php:7.4-apache
   
   ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
   
   RUN \
       # Set -ex and chain commands using ';' instead of '&&'.
       set -ex; \
       \
       apt-get update; \
       \
       chmod +x /usr/local/bin/install-php-extensions; \
       \
       install-php-extensions mysqli
   ```
2. I created an image with: `docker build -t test:latest .`
3. I created a directory on my PC with an `index.php` file containing `<?php phpinfo();`
4. I launched it with
   ```sh
   docker run --rm -p 8000:80 -v path/to/the/test/directory:/var/www/html test:latest
   ```
5. I browsed to `http://localhost:8000`, and here's what I see:
   ![immagine](https://user-images.githubusercontent.com/928116/173112793-85ed1049-8940-4582-a0cd-9ac71638ed59.png)

So, everything seems to work just fine...

**Reply by adil-azmy** [reply]
Thank you for your help.
After knowing it works for you, i tried now with  `--force-rm --no-cache` and it works.

I think that something was messing things up in the cache maybe.
